Applications/Spire/Include/Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorContextMenu.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef SPIRE_ORDER_IMBALANCE_INDICATOR_CONTEXT_MENU_HPP
+#define SPIRE_ORDER_IMBALANCE_INDICATOR_CONTEXT_MENU_HPP
+#include <QMenu>
+#include "Spire/Spire/Spire.hpp"
+
+namespace Spire {
+
+  //! Displays a context menu for the order imbalance indicator window.
+  class OrderImbalanceIndicatorContextMenu : public QMenu {
+    public:
+
+      //! Signals that a menu option was selected.
+      using SelectedSignal = Signal<void ()>;
+
+      //! Signals that a column check box was toggled.
+      /*
+        \param is_checked True if the check box is in the checked state.
+      */
+      using ToggledSignal = Signal<void (bool is_checked)>;
+
+      //! Constructs an OrderImbalanceINdicatorContextMenu.
+      /*
+        \param parent The parent widget.
+      */
+      OrderImbalanceIndicatorContextMenu(QWidget* parent = nullptr);
+
+      //! Connects a slot to the export table signal.
+      boost::signals2::connection connect_export_table_signal(
```

Comment:
Why do these methods have to be `const`?

Reply:
It's a convention I've been following, I'd ask Kamal if you wanted a full answer.

Reply:
As a coding convention, connecting to signals is a ```const``` operation. It allows us to pass ```const``` models that can be listened to for changes without being mutated.